detection_rules/integrations.py

```diff
@@ -0,0 +1,127 @@
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License
+# 2.0; you may not use this file except in compliance with the Elastic License
+# 2.0.
+
+"""Integration functions."""
+import gzip
+import json
+import os
+import re
+
+import yaml
+
+from .ghwrap import GithubClient
+from .semver import Version
+from .utils import INTEGRATION_RULE_DIR, get_etc_path
+
+
+class IntegrationPackages():
+
+    def __init__(self):
+        ...
+
+    @staticmethod
+    def build_integrations_manifest(token: str, overwrite: bool) -> None:
+        """Builds a new local copy of manifest.yaml from integrations Github"""
+
+        if overwrite:
+            manifest_file_path = get_etc_path('integration-manifests.json.gz')
+            if os.path.exists(manifest_file_path):
+                os.remove(manifest_file_path)
+        rule_integrations = os.listdir(INTEGRATION_RULE_DIR)
+        if "endpoint" in rule_integrations:
+            rule_integrations.remove("endpoint")
+
+        final_integration_manifests = dict()
+        for ints in rule_integrations:
+            integration_manifest = get_integration_packages(token=token, org="elastic", repo="package-storage",
+                                                            branch="production", folder=f"packages/{ints}")
+            final_integration_manifests.setdefault(ints, integration_manifest[ints])
+
+        manifest_file = gzip.open(manifest_file_path, "w+")
```

Comment:
if `overwrite` is `False` then `manifest_file_path` will be referenced before set